daemon/logger/awslogs/cloudwatchlogs_test.go

```diff
@@ -691,3 +691,47 @@ func TestCollectBatchWithDuplicateTimestamps(t *testing.T) {
 		}
 	}
 }
+
+func TestAWSLogsNameTemplate(t *testing.T) {
+	for i, test := range []struct {
+		Context logger.Context
+		Result  string
+		Error   error
+	}{
+		{logger.Context{Config: map[string]string{regionKey: "us-east-1", logStreamTemplateKey: "plaintext"}}, "plaintext", nil},
+		{logger.Context{Config: map[string]string{regionKey: "us-east-1", logStreamTemplateKey: fmt.Sprintf("{{ index .Config %q }}", regionKey)}}, "us-east-1", nil},
```

Comment:
No strong opinion here - I felt like the "%q" quoting was more clear than explicit string joins, but again - no objection if you don't like fmt